drivers/source/SFDP.cpp

```diff
@@ -134,5 +174,127 @@ int sfdp_parse_headers(Callback<int(bd_addr_t, void *, bd_size_t)> sfdp_reader,
     return 0;
 }
 
+int sfdp_parse_sector_map_table(Callback<int(bd_addr_t, void *, bd_size_t)> sfdp_reader, sfdp_smptbl_info &smptbl)
+{
+    uint8_t sector_map_table[SFDP_BASIC_PARAMS_TBL_SIZE]; /* Up To 20 DWORDS = 80 Bytes */
+    uint32_t tmp_region_size = 0;
+    int i_ind = 0;
+    int prev_boundary = 0;
+    // Default set to all type bits 1-4 are common
+    int min_common_erase_type_bits = SFDP_ERASE_BITMASK_ALL;
+
+    int status = sfdp_reader(smptbl.addr, sector_map_table, smptbl.size);
+    if (status < 0) {
+        tr_error("table retrieval failed");
```

Comment:
At some point we tried to get all traces to include the function name which called them. Should we perform a similar exercise while refactoring this code block?

Reply:
That's a good question. Now I just start to wonder if that would be redundant information because we have the knowledge about trace group already...

Reply:
I meant something like this: https://github.com/ARMmbed/mbed-os/pull/11460
Trace group will only reveal the "module" or rather source file, that used it?

Reply:
That was inside one driver but once I start looking are Mbed OS codebase that kind of practise doesn't seem to be common at all. Seems like a too much of my own idea - now when I take a look around - that I would want to spread it outside ESP8266 driver.

```
mbed-os$ find -iname "*.cpp" | xargs grep tr_error\( | awk -F"tr_error" '{print $2}' | sort -u
("[%02d] network error %d", i, recvd);
("[%02d] network error %d", i, sent);
("[%02d] network error %d\n", i, recvd);
("[%02d] network error %d\n", i, sent);
("[%02d] sock.send return size %d does not match the expectation %d", i, sent, BURST_SIZE);
("[%02d] sock.send return size %d does not match the expectation %d\n", i, sent, BURST_SIZE);
("[%02d] sock.send return size %d does not match the expectation %d\n", s_idx, sent, pkt_s);
("[%02d] sock.send return size %d does not match the expectation %d", s_idx, sent, pkt_s);
("Accumulated Data (%d) size doesn't match set_start final size (%d) - file: %s", _cur_inc_data_size,
("Allowed FRMPayload = %d, FRMPayload = %d, MAC commands pending = %d",
("APN lookup failed");
("AT3GPP error code %d", get_3gpp_error());
("\"AT+CWCOUNTRY_CUR=%d,\"%s\",%d,%d\" - FAIL", t_ap, country_code, channel_start, channels);
("\"AT+CWCOUNTRY_DEF=%d,\"%s\",%d,%d\" - FAIL", t_ap, country_code, channel_start, channels);
("AT sync failed");
("AT sync failed after PPP Disconnect");
("Attempted to use 4-byte address but 4-byte addressing is not supported");
("Attempt of PPP connect over NON-IP: failed to CONNECT");
("AT trace unable to allocate buffer!");
("BLE TL evt_received, wrong type:%d \r\n");
("[bt#%02d] packet timeout...", i);
("calc_area_params failed - Invalid input addresses");
("Cancelled timeout");
("Cannot set data rate. Please turn off ADR first.");
("CellularContext disconnect failed!");
("Cellular model not found!");
("CellularStateMachine failure: %s", msg);
("CellularStateMachine target state %s, current state %s", get_state_string(_target_state), get_state_string(_state));
("Certificate verification failed:\r\n%s", buf);
("Cinterion model unsupported %s", model);
("_clear_block_protection - Status register read failed");
("__clear_block_protection - Status register write failed");
("Control plane SETUP failed!");
("Couldn't calulate Area Params - err: %d", status);
("Couldn't find valid DeviceKey - err: %d", status);
("Couldn't send a JoinRequest: error %d", status);
("DeviceKey: Fail in mbedtls_platform_setup.");
("DeviceKey: Fail to initialize KvStore configuration.");
("Device not ready after write, failed");
("Device not ready, clearing block protection failed");
("Device not ready, reset failed");
("Device not ready, write failed");
("Device supports Quad bus, but Quad Enable Failed");
("DNS: query \"%s\" => busy", dns_test_hosts[i]);
("DNS: query \"%s\" => busy", hosts[i]);
("DNS: query \"%s\" => DNS failure", dns_test_hosts[i]);
("DNS: query \"%s\" => DNS failure", hosts[i]);
("DNS: query \"%s\" => %d, unexpected answer", hosts[i], err);
("DNS: query \"%s\" => no memory", dns_test_hosts[i]);
("DNS: query \"%s\" => no memory", hosts[i]);
("DNS: query \"%s\" => timeout", dns_test_hosts[i]);
("DNS: query \"%s\" => timeout", hosts[i]);
("erase exceeds flash device size");
("Erase exceeds flash device size");
("[ERROR] Connecting to network. Retrying %d of %d...", tries, MAX_RETRIES);
("[ERROR] Connecting to network. Retrying %d of %d.", tries, MAX_RETRIES);
("Error from sock.connect: %d", err);
("Error from sock.connect: %d\n", err);
("Error from sock.open: %d", err);
("Error from sock.open: %d\n", err);
("Error from sock.set_root_ca_cert: %d\n", err);
("Error from tlssocket_connect_to_echo_srv\n");
("Error: No resources to initiate DNS query for %s", dns_test_hosts[i]);
("Ethernet Driver failed to register with Stack. RetCode=%i", eth_driver_enabled);
("Failed call via eventqueue in do_connect_with_retry()");
("Failed to activate network context! (%d)", err);
("Failed to allocate memory for MLME-SET.request");
("Failed to allocate memory for the msdu");
("Failed to allocate timeout");
("Failed to CONNECT");
("Failed to create socket %d", socket->id);
("Failed to create SW MAC.");
("Failed to enable hw flow control");
("Failed to open data channel!");
("Failed to open socket %d", socket->id);
("Failed to register RF driver.");
("Failed to schedule transmission");
("Failed to set CEREG/CGREG URC's for registration");
("Failed to set PIN: error %d", _cb_data.error);
("Failed to start thread");
("fast mode enable - Writing Config Register failed");
("Fatal Error, Radio failed to transmit");
("File: %s, Exists but write protected", _full_path_key);
("File Verification failed, status: %d", status);
("FlashIAP init failed - err: %d", status);
("FSST Set set_add_data Failed: %d", status);
("FSST Set set_finalize Failed: %d", status);
("FSST Set set_start Failed: %d", status);
("Global block protection unlock failed");
("Heap error (%d), app_heap_error_handler()", event);
("Illegal application port definition.");
("Init - clearing block protection failed");
("Init - Could not read vendor id");
("Init - Detecting/enabling 4-byte addressing failed");
("Init - Detecting erase types instructions/sizes failed");
("Init - Detecting reset protocol/resetting failed");
("init - _is_mem_ready Failed");
("Init - _is_mem_ready Failed");
("init - Parse Basic Param Table Failed");
("Init - Parse Basic Param Table Failed");
("init - Parse Sector Map Table Failed");
("Init - Parse Sector Map Table Failed");
("init - Parse SFDP Headers Failed");
("Init - Parse SFDP Headers Failed");
("init - Read SFDP First Table Failed");
("Init - Read SFDP First Table Failed");
("init - Read Vendor ID Failed");
("init - Unable to initialize flash memory, tests failed");
("init - verify 3byte addressing Failed");
("Init - verify flash density failed");
("Invalid Data Buf Argument");
("invalid erase - unaligned address and size");
("Invalid erase - unaligned address and size");
("Invalid frame size");
("Invalid MAC command (0x%X)!", payload[mac_index]);
("Invalid send flags");
("_is_mem_ready FALSE");
("_is_mem_ready FALSE: status value = 0x%x ", status_value);
("Joining abandoned: CRYPTO_ERROR");
("Joining abandoned: Radio failed to transmit");
("KV Config: Can not create TDBStore with less then 2 sector.");
("KV Config: DataFlashBlockDevice init fail");
("KV Config: End address is not aligned. Consider changing the size parameter.");
("KV Config: End address is out of boundaries");
("KV Config: Fail to attach KvStore main instance to KVStore global API");
("KV Config: Fail to attach KVStore main instance to KVStore global API.");
("KV Config: Fail to deinit internal BlockDevice.");
("KV Config: Fail to get addresses for SlicingBlockDevice.");
("KV Config: Fail to get external BlockDevice ");
("KV Config: Fail to get external BlockDevice.");
("KV Config: Fail to get FileSystem");
("KV Config: Fail to get FileSystemStore");
("KV Config: Fail to get internal BlockDevice ");
("KV Config: Fail to get internal BlockDevice.");
("KV Config: Fail to init external BlockDevice ");
("KV Config: Fail to init external BlockDevice.");
("KV Config: Fail to init internal BlockDevice ");
("KV Config: Fail to init internal BlockDevice.");
("KV Config: Fail to init internal TDBStore.");
("KV Config: Fail to init internal TDBStore");
("KV Config: Fail to init KVStore global API.");
("KV Config: Fail to init KVStore global API");
("KV Config: Fail to init SecureStore.");
("KV Config: Fail to mount FileSystem to %s",
("KV Config: Internal block device end address is out of boundaries");
("KV Config: Internal block device start address is not aligned. Better use %02llx", aligned_start_address);
("KV Config: Internal block device start address is not aligned. Consider changing the size parameter");
("KV Config: Internal block device start address overlapped ROM address ");
("KV Config: No default component define in target.json for this target.");
("KV Config: \"other\" block device init fail");
("KV Config: QSPIFBlockDevice init fail");
("KV Config: SDBlockDevice init fail");
("KV Config: SPIFBlockDevice init fail");
("KV Config: Start address is not aligned. Better use %02llx", aligned_start_address);
("KV Dir: %s, dir_close failed", _cfg_fs_path); //TBD verify ERRNO NOEXIST
("KV Dir: %s, doesnt exist", _cfg_fs_path); //TBD verify ERRNO NOEXIST
("KV Dir: %s, mkdir failed.. ", _cfg_fs_path); //TBD verify ERRNO NOEXIST
("Legacy 4k erase instruction not supported");
("LoWPANND initialize failed: %d", result);
("MIC failed");
("Modem not responding");
("Must assign a callback function for link check request. ");
("NanostackSocket::accept() failed");
("NanostackSocket::open() failed");
("network error %d, missing %d bytes from a burst\n", recvd, bt_left);
("network error %d, missing %d bytes from a burst", recvd, bt_left);
("network error %d", recvd);
("network error %d", sent);
("NIDD connection open failed with error: %d", err);
("No cellular stack!");
("No erase type was found for current region addr");
("no erase type was found for region addr");
("No erase type was found for region addr");
("No free memory for timeout");
("No free memory for timeout equeue");
("No free timeouts");
("No optval!");
("no region found for address %llu", addr);
("No socket found to be closed");
("Not received as much data as expected. Should receive: %d bytes, received: %d bytes", recv_len, hexlen);
("nsapi_ppp_connect failed");
("_oob_busy(): unrecognized busy state '%c...'", status);
("_oob_connection_status(): Invalid AT cmd \'%s\' .", status);
("_oob_connection_status(): Network status timeout, disconnecting.");
("Param Header: - Major Version should be 1!");
("PIN required but no SIM pin provided.");
("[pkt#%02d] network error %d", i, recvd);
("[pkt#%02d] network error %d\n", i, recvd);
("QSPI After Erase Device not ready - failed");
("_qspi_configure_format failed");
("QSPI Erase command failed!");
("QSPI Erase Device not ready - failed");
("QSPI Erase failed");
("QSPI Erase - Updating 4-byte addressing extended address register failed");
("QSPIFBlockDevice with the same csel(%d) already exists", (int)_csel);
("QSPIFBlockDevice with the same csel(%d) already exists", (int)csel);
("QSPI Generic command - Updating 4-byte addressing extended address register failed");
("QSPI Read failed");
("QSPI Read - Updating 4-byte addressing extended address register failed");
("QSPI Set Frequency Failed");
("QSPI Write failed");
("QSPI Write - Updating 4-byte addressing extended address register failed");
("QUECTEL_M26_CellularStack:%s:%u:[NSAPI_ERROR_DEVICE_ERROR]", __FUNCTION__, __LINE__);
("QUECTEL_M26_CellularStack:%s:%u:[NSAPI_ERROR_NO_SOCKET]", __FUNCTION__, __LINE__);
("QUECTEL_M26_CellularStack:%s:%u:[NSAPI_ERROR_PARAMETER]", __FUNCTION__, __LINE__);
("QUECTEL_M26_CellularStack:%s:%u:[TCP: %d]", __FUNCTION__, __LINE__, error);
("Read Command failed");
("Reading Config Register failed");
("Reading Status Register 1 failed");
("Reading Status Register 2 failed");
("Reading Status Register failed");
("Read Vendor ID Failed");
("Received unsupported data coding scheme: 0x%02x", scheme);
("_recv_tcp_passive(): Unknown state.");
("Requesting stack for NON-IP context! Should request control plane netif: get_cp_netif()");
("Retries exhausted for Class %s device",
("retrieving Parameter Header %d failed", i_ind + 1);
("retrieving SFDP Header failed");
("RM1000_AT_CellularStack::create_socket_impl error sock_id=%d err=%d", sock_id, err);
("[Round#%02d] network error %d", i, recvd);
("[Round#%02d] network error %d\n", i, recvd);
("[Round#%02d] network error %d\n", s_idx, recvd);
("[Round#%02d] network error %d\n", s_idx, sent);
("[Round#%02d] network error %d", s_idx, recvd);
("[Round#%02d] network error %d", s_idx, sent);
("[round#%02d] unable to open new socket, error: %d", i, ret);
("[round#%02d] unable to open new socket, error: %d\n", i, ret);
("RX_ERROR: mcps_indication status = %d", mcps_indication->status);
("Sector Map - Supporting Only Single! Map Descriptor (not map commands)");
("[Sender#%02d] network error %d\n", s_idx, sent);
("[Sender#%02d] network error %d", s_idx, sent);
("send: err %d, time %d", err, timer.read_ms());
("Sending Generic command: %x", instruction);
("Sending RSTEN failed");
("Sending RST failed");
("Sending SFDP read instruction");
("Sending WREN command FAILED");
("sendto size %d (max %d)", size, UDP_PACKET_SIZE);
("_set_write_enable failed");
("_set_write_enable failed - status register 1 value: %u", status_value);
("_sfdp_set_qpi_enabled - send command 35h Failed");
("_sfdp_set_qpi_enabled - send command 38h Failed");
("_sfdp_set_qpi_enabled - send command 61 Failed");
("_sfdp_set_qpi_enabled - send command 65h Failed");
("_sfdp_set_qpi_enabled - send command 71h Failed");
("_sfdp_set_qpi_enabled - set config bit 6 command 65h Failed");
("shci_cmd_resp_wait timed out\r\n");
("SIM PUK required");
("Socket create failed!");
("Socket %d create error %d", find_socket_index(socket), ret_val);
("Socket %d create %s error %d", find_socket_index(socket), addr.get_ip_address(), ret_val);
("Socket %d not responding", socket->id);
("Socket %d open failed!", internet_service_id);
("Socket %d recv error %d", find_socket_index(socket), ret_val);
("Socket %d recvfrom addr (len %d)", socket->id, ip_len);
("Socket %d recvfrom failed!", socket->id);
("Socket %d send failed", socket->id);
("Socket %d send failure", socket->id);
("Socket %d sendto %s error %d", find_socket_index(socket), addr.get_ip_address(), ret_val);
("Socket id %d != %d", socket_id, socket->id);
("Socket not found %d", sock_id);
("Socket recv error %d", recv);
("Socket recvfrom id %d != %d", socket_id, socket->id);
("Socket send error %d", size);
("socket_sendmsg: error=%d", retcode);
("socket_sendto() not supported with TCP!");
("sock.recv returned an error %d", recvd);
("SPI After Erase Device not ready - failed");
("SPI Erase Device not ready - failed");
("_spi_send_read_sfdp_command failed");
("SPI Set Frequency Failed");
("Start state machine failed.");
("Status register not set correctly");
("Supporting up to %d regions, current setup to %d regions - fail",
("table retrieval failed");
("Thread initialize failed: %d", result);
("Timeout already set");
("Too many different QSPIFBlockDevice devices - max allowed: %d", QSPIF_MAX_ACTIVE_FLASH_DEVICES);
("Trying to set non-TCPSocket as TLSSocket");
("Unknown message type!");
("Unknown MLME Indication type.");
("Unknown SIM state %s", simstr);
("Unknown state: %d:", new_state);
("Unsupported sockopt (%d)", optname);
("Upload error! orig = %d, uploaded = %d", size, upload_size);
("User defined port in .json is illegal.");
("Verifying Config Register failed");
("verify SFDP signature and version Failed");
("Wi-SUN initialize failed: %d", result);
("Write Disable failed");
("Write Enabe failed");
("Write enable failed");
("Write Enable failed");
("Write failed");
("Writing Status Register 1 failed");
("Writing Status Register 2 failed");
("Writing Status Registers failed");
```